responses/matchers.py

```diff
@@ -77,11 +87,40 @@ def match(request):
         )
 
         if not valid:
-            return False, "%s doesn't match %s" % (
+            return False, "Query parameters do not match. %s doesn't match %s" % (
                 sorted(request_params.items()),
                 sorted(params.items()),
             )
 
         return valid, ""
 
     return match
+
+
+def header_matcher(headers, allow_additional=True):
+    """
+    Matcher to match 'headers' argument in request using the responses library.
+    :param headers: (dict), same as provided to request
+    :param allow_additional: (bool), ignore and allow additional headers that
```

Comment:
may be call argument `strict_match` and set default to `False`
may be it is just my personal, but for me that sounds more explicit and easier to understand

then you check
~~~python
if not strict_match:
    for header in headers:
        if header not in request.headers:
            return False
~~~


~~~python
if strict_match:
    sorted(headers) == sorted(request.headers)
~~~

Reply:
I really struggled to come up with a good name here, and I'd agree that yours is better. I'll make this change.